### PR TITLE
add public mint chain

### PR DESCRIPTION
**Pull request by zaidhub**
Is the icon obtained from the favicon of the "infoURL"?

### CODE REVIEW

**Review by Portalssecurity — APPROVED** [review]
gh pr checkout 1120 #1104 zaidhub:adding-public-mintsrc/tokens/go/0x67bBB47f6942486184f08a671155FCFA6cAd8d71.jsonimport eth_utils


def checksum_encode(addr): # Takes a 20-byte binary address as input
    hex_addr = addr.hex()
    checksummed_buffer = ""

    # Treat the hex address as ascii/utf-8 for keccak256 hashing
    hashed_address = eth_utils.keccak(text=hex_addr).hex()

    # Iterate over each character in the hex address
    for nibble_index, character in enumerate(hex_addr):

        if character in "0123456789":
            # We can't upper-case the decimal digits
            checksummed_buffer += character
        elif character in "abcdef":
            # Check if the corresponding hex digit (nibble) in the hash is 8 or higher
            hashed_address_nibble = int(hashed_address[nibble_index], 16)
            if hashed_address_nibble > 7:
                checksummed_buffer += character.upper()
            else:
                checksummed_buffer += character
        else:
            raise eth_utils.ValidationError(
                f"Unrecognized hex character {character!r} at position {nibble_index}"
            )

    return "0x" + checksummed_buffer


def test(addr_str):
    addr_bytes = eth_utils.to_bytes(hexstr=addr_str)
    checksum_encoded = checksum_encode(addr_bytes)
    assert checksum_encoded == addr_str, f"{checksum_encoded} != expected {addr_str}"


test("0x5aAeb6053F3E94C9b9A09f33669435E7Ef1BeAed")
test("0xfB6916095ca1df60bB79Ce92cE3Ea74c37c5d359")
test("0xdbF03B407c01E7cD3CBea99509d93f8DDDC8C6FB")
test("0xD1220A0cf47c7B9Be7A2E6BA89F429762e7b9aDb")

File: _data/chains/eip155-20201.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "symbol": "USD",
+    "decimals": 18
+  },
+  "infoURL": "https://publicmint.com/",
+  "shortName": "mint",
+  "chainId": 2020,
+  "networkId": 2020,
+  "explorers": [{
+    "name": "Public Mint Explorer",
+    "url": "https://explorer.publicmint.io/",
```
Comment (ligi) [inline _data/chains/eip155-20201.json:20]:
```suggestion
    "url": "https://explorer.publicmint.io",
```

**Review by ligi — CHANGES_REQUESTED** [review]
The ChainID in the filename MUST match the one in the json

**Review by Raines333 — APPROVED** [review]

**Comment by zaidhub** [comment]
> The ChainID in the filename MUST match the one in the json

Hi, there's already another filename with our ChainID. The other file is not ours and it's not related to us. Apparently it's not even activated. What can we do in these cases? We use ChainID 2020 for mainnet and 2019 for testnet.

**Comment by zaidhub** [comment]
Hi, any response?
Should I just submit a new PR with the filename correctly?

**Comment by Raines333** [comment]
What

**Review by Raines333 — APPROVED** [review]

**Comment by Raines333** [comment]
Yeah I’d just submit new PR with file name correctly

**Comment by Raines333** [comment]
Bro I’ve just been trying to teach myself you’re the first person to email
me. I’m to smart for my own good but no friends that mess with code.. I’ve
been buying btc since 2011 and silk rd but minting and running codes and
nodes.  Just trying to get some knowledge and education/friends through
these platforms ya knkw just learn man. So ya ever need a kick in stick or
to vent about these codes I’m all ears

**Pull request closed without merging**